File: app/src/main/java/me/saket/dank/ui/PlaygroundActivity.java
package me.saket.dank.ui;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.PopupMenu;
import android.widget.Button;
import butterknife.ButterKnife;
import butterknife.OnClick;
import me.saket.dank.R;
import me.saket.dank.data.RedditLink.User;
import me.saket.dank.ui.user.UserProfilePopup;
import me.saket.dank.utils.UrlParser;

public class PlaygroundActivity extends DankPullCollapsibleActivity {

  public static void start(Context context) {
    context.startActivity(new Intent(context, PlaygroundActivity.class));
  }

  @Override
  protected void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_playground);
    ButterKnife.bind(this);
    findAndSetupToolbar();

    setupContentExpandablePage(ButterKnife.findById(this, R.id.playground_root));
    expandFromBelowToolbar();
  }

  @OnClick(R.id.playground_user_profile_popup)
  void onClickShowProfilePopup(Button button) {
    UserProfilePopup userProfilePopup = new UserProfilePopup(this);
    userProfilePopup.loadUserProfile((User) UrlParser.parse("http://reddit.com/u/saketme"));
    userProfilePopup.show(button);
  }

  @OnClick(R.id.playground_menupopup)
  void onClickShowMenuPopup(Button button) {
    PopupMenu popupMenu = new PopupMenu(this, button);
    popupMenu.getMenu().add("Poop");
    popupMenu.getMenu().add("Poop");
    popupMenu.getMenu().add("Poop");
    popupMenu.show();
  }

}

File: app/src/main/java/me/saket/dank/ui/PlaygroundActivity.kt
package me.saket.dank.ui

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.widget.PopupMenu
import butterknife.ButterKnife
import butterknife.OnClick
import me.saket.dank.R
import me.saket.dank.di.Dank
import me.saket.dank.ui.user.UserProfilePopup
import me.saket.dank.urlparser.RedditUserLink
import me.saket.dank.urlparser.UrlParser
import javax.inject.Inject

class PlaygroundActivity : DankPullCollapsibleActivity() {

  @Inject
  lateinit var urlParser: UrlParser

  override fun onCreate(savedInstanceState: Bundle?) {
    Dank.dependencyInjector().inject(this)
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_playground)
    ButterKnife.bind(this)
    findAndSetupToolbar()

    setupContentExpandablePage(findViewById(R.id.playground_root))
    expandFromBelowToolbar()
  }

  @OnClick(R.id.playground_user_with_subreddit)
  internal fun onClickShowProfilePopup1(button: Button) {
    showPopup(button, "http://reddit.com/u/kn0thing")
  }

  @OnClick(R.id.playground_banned_user)
  internal fun onClickShowProfilePopup2(button: Button) {
    showPopup(button, "http://reddit.com/u/TheFlintASteel")
  }

  @OnClick(R.id.playground_non_existent_user)
  internal fun onClickShowProfilePopup3(button: Button) {
    showPopup(button, "http://reddit.com/u/Sak3tme")
  }

  @OnClick(R.id.playground_deleted_user)
  internal fun onClickShowProfilePopup4(button: Button) {
    showPopup(button, "http://reddit.com/u/mosnuk123")
  }

  private fun showPopup(button: Button, username: String) {
    val userProfilePopup = UserProfilePopup(this)
    userProfilePopup.loadUserProfile(urlParser.parse(username) as RedditUserLink)
    userProfilePopup.showWithAnchor(button)
  }

  @OnClick(R.id.playground_menupopup)
  internal fun onClickShowMenuPopup(button: Button) {
    val popupMenu = PopupMenu(this, button)
    popupMenu.menu.add("Poop")
    popupMenu.menu.add("Poop")
    popupMenu.menu.add("Poop")
    popupMenu.show()
  }

  companion object {

    fun start(context: Context) {
      context.startActivity(Intent(context, PlaygroundActivity::class.java))
    }
  }
}
